### ISSUE TITLE
B027 false positive for .pyi files

### ISSUE THREAD

**Issue by JonathanPlasse** [charliermarsh/ruff#3012]
```python
from _typeshed import Incomplete
from abc import ABC, abstractmethod

class _ProgressColumn(ABC):
    def __init__(self, table_column: Incomplete | None = ...) -> None: ...
    def get_table_column(self) -> Incomplete: ...
    def __call__(self, task: Incomplete) -> Incomplete: ...
    @abstractmethod
    def render(self, task: Incomplete) -> Incomplete: ...
```

`ruff` detect the `B027` (is an empty method in an abstract base class, but has no abstract decorator) however this rule should not be activated with `.pyi` files where every method should be empty.

[Link to the playground](https://play.ruff.rs/#N4KABGBECGA2sHsDuBTAJgWgMYIHYDMBXAZ2gCNYVjIAuMAbQF0AacKMwgS1gBdPdqdJqwiQ0hALYSAnhgBu0AE6dylDIoDmAD1pQAegAoA+gGoAPsZP1oGAF4BBDAC0ADBgCcRxgCprdx64ejCYA-ACUYQAkkCJQKFo8KLiYnBq4CIoousJskPGJyRjEKJRYPNksufkoirhwFbGQqemZDbmw-CgYlLgaPAAWugAcQ43FpeVCbKIAQjHTUACi8xCiAGKQbJWiPEoaKDzyNcSceLqQAA7SAMwAjC4rUPiw0ADWKEMY0Ljpu3x4gjAoFWUBkVww-E4h0yPEItV0+DgxViomIhAuF0yxGIGHEUlke0BiNgyIWkDRGKxOPSuC6MLhuH4GgRSJQKKgcEQSCKu0UX00X1w0hZJLZZOaGS6RHgskIuB40gu6BFxTYAF9Gs83h8MGRvmgobpgatIP0lGgcGh0BgeBILrjOJkyhlhVMQaIAPS2i6PEGQD0KRReu2+k0eq1yD3EfoSTYg7YmrD9FBYV42xXW+JYFAXf64FUodWal7vT4cDRkFBKI1k6qFThSQi7ChdLCcwFMIu5LWl3VcXj8QHG0Qcbh8AQQtKSjrESYMRhd0Q9nU1RQSYjMujD0HQLQ85S9bpJPqDOguRdPEs6gCOhAQiSH4twHVpGFv96o5zQCEILdDkAkQgBxfLp3wfL8fz-dkxAQLBZwPDQ3zvcC6Bg39KH-aA5AQThMCoNslV0HhFEIQsIA1bsr0+cZ8BrP0JUyDA6gkT83XdSAjAkbVQ1EIxoGIA0yh4qAjEyC4XmzYSOPwTgSjQah2V4mS5KMK1ESAnhqAWBdyOLbVPj4NBZAbC4Mk0uiTT1XB1BKaA+DkLoTLMwFLiUJJzOgqzaUwaALk4I01QvSBlwMjNsGTVMmQs1FiM4IS6GJUk-XiFA7UOCQEHESgOwWHZFSi7SgpCiE7Q6LAoX3bA8DbOdtxgeBkAwQDgM6IiSLIsAKKXKiStMxRDhwXAHPlU4BGijkOn41igVyqALn1fjzguNApKuJRFGQJboFWxAeFgaQADp4kxKgXIuHRFKgYpyAyfNUOIMbLsgApiAyZ5NtQnhaKe3BJCuc5cB9J6uJ4cT7w6MhzlKqSQbBvbODIA6rjhpbeFWhAXkUM7YFW74SiW-MnrgXZHXOYmpP6DGEDkWSkBc-o5DjVZAt0yj9IwK4H0OWdpAwrcyRkhI4S6ebMnlZNikBYjSOg0XoBY2LbC6ZiqHTQjPvRDDZaUeWDmUJX5DgUicQVdWoBncptcUXXFa6BRYGN9RGtNrINfErJoOtzhJZs29HS6EGkwwfAMmyWbIBmKbFi0bNc1GqTo9jvMpIANSNlBFkUDbFCkgB5ABlTPs6kgBJXOi9Dp7FiGx08BY+UK5zp7XtTA4jqz0PtM96BvdVutMEyP3GMD-pg9DoQE1ELjFDTUX3Il6bpY6rrL3ZuUSGtPZJHcx96KnRjAxUQSmN1olWSC72zPG4KMmzDAkGt+04BUSWC2gkPFDvk5ejUEC37Jb+Ghf6dAwFmB2JwARtD9DgCQZAQH8V6s5f+fpiDiQqngG0XsXyIRgVxNqMsyQZCtHyMgsgXb4LFH6D+X8zL3yhP0fgRQUx5jPqKd+t8ug8AQDaBAPoJ7QVeOkJA1kZJY0OKLBUUCTSCOQNZAYjpMASNdPOARQjrKIDbLAMesBiFSNEPka2PIFqKEwBDa2ihlE5D9JkF49lHJ2mchgIhNRzhEH6hLQ4XDsCIElpbMkg8uCZBSA4-qHZJ5QCwC8bE00rGJgBLseUYToKH1UDE8JkB0jdE6DiSsH9XYqLJCBHE0B8CJD5E5UJugMC3GgkU3UBxUBJDVtNFw7DP7WmKHLRIFQgoSCwG2Ss18uJ7hge7LQUJlH3CCkqC4nxmJRX5nvFoytT5h3dFdA4ABVIG4dEhKAACKyKksUHg2yADCUSFK7KrIoQ5wiLn8Sues8kWyLgAFlMpAQ9tcg5siPlZW+c8-i0hcBYHzq8qSwLQUABUbl3MJuHE52zYWzn2XZbaT1ETcGFonHMycnqIF6G8060B9gw13Pszg+BaJdzJJEx5CtKaYCtDga2XCsZrPYvS7EjLMpMwgOk2cdk4q8uZSmDIdkMg5WeUKvgWBRX8rADpTqjQriWioAqPmM0lmSicQ5RQhLEK7GIGmXBO8CzTOkN+OCmr8nABXpcXm-Bapkk5I1LiGg4qGwdpwjM0r2LwSkqQ8CtK-TDN1NbUFEtdC3AAEzQXDfSWogIABsCbdz8g0ICAArOmvcsrUrmroNm88rNRBXHRBoa2Vpr7vBzOoOUfAWJqwWWARKy8QBqkiN2-AG0JBgCMC7aM6AwAVJ4GAEuoKEClQOIWXt06wDkCwKOkJ477AzDOcwRdZB4LQDKKKkAIBuXEAHQABQ2lW06ZyMaSFwAYddZywg0AWGpAdRhIQ8CMEYAwNEt3NkoEYHADsJC4DoJO0ZlBEhgDMGAAAcngFAYAAC8YADpobCGADAAA+ODCG6BoYOi+lA+AwD7E-f+lAgGb0gZ-SUfAGHsMTqnTOxI+G0NEZI1+zRsAv20dgPgP9-FXhgeY+7RIDGcPgenWJlAbHCMggAALkF3fug4TKONgDFsQvjAmwDGuE0xiDs6JOGek5B2TqH2NAA)
